File: app/src/main/java/io/ermdev/classify/di/component/DatabaseComponent.java
package io.ermdev.classify.di.component;

import dagger.Component;
import io.ermdev.classify.di.module.AppModule;
import io.ermdev.classify.di.module.DatabaseModule;
import io.ermdev.classify.ui.main.MainPresenter;

/**
 * Created by erafaelmanuel on 11/28/2017.
 */

@Component(modules = {AppModule.class, DatabaseModule.class})
public interface DatabaseComponent {

    void inject(MainPresenter mPresenter);
}

File: app/src/main/java/io/ermdev/classify/di/component/DatabaseComponent.kt
package io.ermdev.classify.di.component

import dagger.Component
import io.ermdev.classify.di.module.AppModule
import io.ermdev.classify.di.module.DatabaseModule
import io.ermdev.classify.ui.main.MainActivity
import javax.inject.Singleton

/**
 * Created by erafaelmanuel on 1/3/2018.
 */

@Singleton
@Component(modules = arrayOf(AppModule::class, DatabaseModule::class))
interface DatabaseComponent {
    fun inject(activity: MainActivity)
}